### ISSUE TITLE
Build with Visual Studio based on .NET Core/5.0

### ISSUE THREAD

**Issue by KarloX2** [industrial-edge/pingpong-csharp#6]
Hello,

I'm wondering if this project can be built with Microsoft tools, preferably Visual Studio?

And, can I build and run it using .NET Core (or the latest, .NET 5)?

Thanks

**Comment by Boehmi-TB** [comment]
Thanks for raising this issue we will take a look at it

**Comment by EvaKaul** [comment]
Hello @KarloX2 ,
1) It is also possible to integrate this [source file ](https://github.com/industrial-edge/pingpong-csharp/blob/main/src/PingPong.cs) into a Visual Studio project and build it there. The needed references (M2Mqtt, Newtonsoft) can be installed as NuGet packages:

![image](https://user-images.githubusercontent.com/74291028/133411573-d4abefd0-c25a-458f-ae38-3397deaeb6d9.png)

2) This example was implemented and tested only with Mono under Docker, but it should also be possible using .NET

**Comment by KarloX2** [comment]
Thanks, @EvaKaul.

Can you please also tell which project type (in Visual Studio) you recommend?

**Comment by EvaKaul** [comment]
Hi @KarloX2 ,
I choosed a C# Console Application in VS to build the source code.
But to test the whole ping pong application you have to use an Industrial Edge Device, where the Databus (mqtt broker) is running and config parameters are available.

**Comment by KarloX2** [comment]
Hi @EvaKaul.

Thanks, I am trying that right now. I have selected .NET5 as the target platform. After adding the two nuget packages you mentioned, the code compiles, but I get this warning:

`warning NU1701: Package 'M2Mqtt 4.3.0' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework 'net5.0'. This package may not be fully compatible with your project.
`

So this might be a problem when publishing the code to the IE Device. I'd like to publish it along with .NET5 (self-contained), but legacy .NET Framework (4.x etc.) won't be there, right? Are you sure the M2Mqtt package is the right one to use in this case?

**Comment by EvaKaul** [comment]
reopened

**Comment by EvaKaul** [comment]
Hi @KarloX2 ,
I think I didn't got you right.

Building ONLY the C# solution in VS - without any docker extension - worked for me with my default settings (.NET Framework 4.5.2) and with the latest versions of the nuget packages. I got no warnings.
![image](https://user-images.githubusercontent.com/74291028/133449558-3e5e7046-6ef5-4976-b705-b1ef8a6debdc.png)
![image](https://user-images.githubusercontent.com/74291028/133449686-8b85a91a-0277-4093-b8c8-e270e201387e.png)

But you want to implement the whole docker application with Visual Studio and .NET 5, right? I did not try this, but for sure this is possible using the right VS docker extensions.
